src/string/grapheme/properties.cr

```diff
@@ -0,0 +1,1670 @@
+module String::Grapheme
+  # The unicode properties. Only the ones needed in the context of this shard are included.
+  private enum Property
+    Any
+    Preprend
+    CR
+    LF
+    Control
+    Extend
+    RegionalIndicator
+    SpacingMark
+    L
+    V
+    T
+    LV
+    LVT
+    ZWJ
+    ExtendedPictographic
+
+    # returns the Unicode property value (see `Codepoints` constants below)
+    # of the given code point
+    def self.from(r : Int32)
+      # run a binary search
+      f = 0
+      t = Grapheme.codepoints.size
+      while t > f
+        mid = (f + t) // 2
+        cp = Grapheme.codepoints[mid]
+        if r < cp[0]
+          t = mid
+          next
+        end
+        if r > cp[1]
+          f = mid + 1
+          next
+        end
+        return cp[2]
+      end
+      Property::Any
+    end
+  end
+
+  # Maps code point ranges to their properties. In the context of this package,
+  # any code point that is not contained may map to "prAny". The code point
+  # ranges in this slice are numerically sorted.
+  #
+  # These ranges were taken from
+  # http://www.unicode.org/Public/UCD/latest/ucd/auxiliary/GraphemeBreakProperty.txt
+  # as well as
+  # https://unicode.org/Public/emoji/latest/emoji-data.txt
```

Comment:
This link doesn't seem to work.  Maybe one of these? https://unicode.org/Public/emoji/13.1/

Reply:
Or maybe https://unicode.org/Public/13.0.0/ucd/emoji/emoji-data.txt